backend/api_server/app.py

```diff
@@ -23,6 +25,35 @@
 configure_logging(APP_NAME)
 
 
+def should_configure_datadog_tracing():
+    return os.environ.get("DD_AGENT_HOST", None) and os.environ.get("DD_TRACE_AGENT_PORT", None)
+
+
+if should_configure_datadog_tracing():
+    # Datadog APM tracing
+    # See https://ddtrace.readthedocs.io/en/stable/basic_usage.html#patch-all
+
+    tracer.configure(
+        hostname=os.environ["DD_AGENT_HOST"],
+        port=os.environ["DD_TRACE_AGENT_PORT"],
+        # Filter out health check endpoint (index page: '/')
+        settings={
+            "FILTERS": [
+                FilterRequestsOnUrl([r"http://.*/$"]),
+            ],
+        },
+    )
+    patch_all()
+
+# TODO: Determine if there is a requirement to instrument deeper
+# profiling in the dev environment.
+# see ticket: https://github.com/chanzuckerberg/single-cell-data-portal/issues/5821
+# enable Datadog profiling for development
```

Comment:
nit: newline above? this seems like 2 separate comment blocks; maybe explain what "dd profiling" provides beyond what is already instrumented for APM

Reply:
will remove this comment entirely. If we need deeper profiling, we can add it.